sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -124,6 +124,7 @@ abstract class ConfEntry[T](val key: String, val converter: String => T, val doc
 
   def get(conf: Map[String, String]): T
   def get(conf: SQLConf): T
+  def getDefault(): T
```

Comment:
It would rather make sense to omit the conf withtout defaults in the dump IMO

Reply:
So, I think this API should just return whatever the default is. Any business logic should not be involved at this level.

For `DumpDefaultConfigs`, it dumps even those configs without defaults because we would want to unset them for the integration test if they were set somehow.